### PR TITLE
T343430: WDIO 7 Async

### PR DESCRIPTION
**Pull request by lorenjohnson**

### CODE REVIEW

File: test/specs/fedprops/item.js [hunk 1]
```diff
@@ -24,94 +24,108 @@ describe( 'Fed props Item', function () {
...
 		};
-		browser.call( () => WikibaseApi.createItem( Util.getTestString( itemLabel ), data ) );
+		await WikibaseApi.createItem( Util.getTestString( itemLabel ), data );
 
-		browser.url( process.env.MW_SERVER + '/wiki/Item:' + itemId );
+		await browser.url( process.env.MW_SERVER + '/wiki/Item:' + itemId );
 
-		const actualPropertyValue = $( '.wikibase-statementgroupview-property' ).getText();
+		const actualPropertyEl = await $( '.wikibase-statementgroupview-property' );
+		const actualPropertyValue = await actualPropertyEl.getText();
```
Comment (rti) [inline test/specs/fedprops/item.js:46]:
```suggestion
		const actualPropertyEl = await $( '.wikibase-statementgroupview-property' );
		await actualPropertyEl.waitForDisplayed();
		const actualPropertyValue = await actualPropertyEl.getText();
```

File: test/specs/quickstatements/quickstatements.js [hunk 2]
```diff
@@ -20,105 +20,136 @@ const getQualifierType = function ( response, propertyId, qualPropertyId ) {
...
+		await wpPasswordEl.setValue( process.env.MW_ADMIN_PASS );
+		await wpLoginButtonEl.click();
 
 		// oauth dialog
-		$( '#mw-mwoauth-authorize-form' ).waitForDisplayed();
-		$( '#mw-mwoauth-accept' ).click();
+		const authFormEl = await $( '#mw-mwoauth-authorize-form' );
+		await authFormEl.waitForDisplayed();
+		const authFormAcceptEl = await $( '#mw-mwoauth-accept' );
+		await authFormAcceptEl.click();
```
Comment (rti) [inline test/specs/quickstatements/quickstatements.js:52]:
```suggestion
		const authFormAcceptEl = await $( '#mw-mwoauth-accept' );
		await authFormAcceptEl.waitForDisplayed();
		await authFormAcceptEl.click();
```

File: test/specs/repo/extensions/nuke.js [hunk 3]
```diff
@@ -1,52 +1,51 @@
...
-		browser.acceptAlert();
+		const formEl = await $( 'form li' );
+		await formEl.waitForDisplayed();
 
+		const checkboxEl = await $( '.mw-checkbox-none' );
+		await checkboxEl.click();
+		const vandalismCheckEl = await $( 'input[value="Vandalism"]' );
+		await vandalismCheckEl.click();
+		const submitButtonEl = await $( 'input[type="submit"]' );
+		await submitButtonEl.click();
```
Comment (rti) [inline test/specs/repo/extensions/nuke.js:44]:
```suggestion
		const checkboxEl = await $( '.mw-checkbox-none' );
		await checkboxEl.waitForDisplayed();
		await checkboxEl.click();
		const vandalismCheckEl = await $( 'input[value="Vandalism"]' );
		await vandalismCheckEl.waitForDisplayed();
		await vandalismCheckEl.click();
		const submitButtonEl = await $( 'input[type="submit"]' );
		await submitButtonEl.waitForDisplayed();
		await submitButtonEl.click();
```

File: test/specs/repo/extensions/universal-language-selector.js [hunk 4]
```diff
@@ -4,21 +4,24 @@ const assert = require( 'assert' );
...
 
-		browser.url( process.env.MW_SERVER );
-		$( '#searchInput' ).waitForDisplayed();
-		$( '#searchInput' ).click();
-		$( '.imeselector' ).waitForDisplayed();
-		$( '.imeselector' ).click();
-
-		const firstLang = $( '.imeselector-menu h3' ).getText();
+		const firstLangEl = await $( '.imeselector-menu h3' );
+		const firstLang = await firstLangEl.getText();
```
Comment (rti) [inline test/specs/repo/extensions/universal-language-selector.js:23]:
```suggestion
		const firstLangEl = await $( '.imeselector-menu h3' );
		await firstLangEl.waitForDisplayed();
		const firstLang = await firstLangEl.getText();
```

File: test/specs/repo/extensions/visual-editor.js [hunk 5]
```diff
@@ -4,43 +4,49 @@ const assert = require( 'assert' );
...
 
 		// start editing
-		$( '.oo-ui-messageDialog-actions' ).waitForDisplayed();
-		const startEditbutton = $$( '.oo-ui-messageDialog-actions a' )[ 1 ];
-		startEditbutton.click();
+		const messageDialogEl = await $( '.oo-ui-messageDialog-actions' );
+		await messageDialogEl.waitForDisplayed();
+		const messageDialogAEl = await $$( '.oo-ui-messageDialog-actions a' );
+		const startEditbutton = messageDialogAEl[ 1 ];
+		await startEditbutton.click();
```
Comment (rti) [inline test/specs/repo/extensions/visual-editor.js:19]:
```suggestion
		const startEditbutton = messageDialogAEl[ 1 ];
		await startEditbutton.waitForDisplayed();
		await startEditbutton.click();
```

File: test/specs/repo/extensions/visual-editor.js [hunk 6]
```diff
@@ -4,43 +4,49 @@ const assert = require( 'assert' );
...
 		// disable notice popup and focus on editor
-		$( '.oo-ui-tool-name-notices .oo-ui-tool-link' ).waitForDisplayed();
-		$( '.oo-ui-tool-name-notices .oo-ui-tool-link' ).click();
-		$( '.ve-ce-surface' ).click();
+		const toolLinkEl = await $( '.oo-ui-tool-name-notices .oo-ui-tool-link' );
+		await toolLinkEl.waitForDisplayed();
+		await toolLinkEl.click();
+
+		const surfaceEl = await $( '.ve-ce-surface' );
+		await surfaceEl.click();
```
Comment (rti) [inline test/specs/repo/extensions/visual-editor.js:29]:
```suggestion
		const surfaceEl = await $( '.ve-ce-surface' );
		await surfaceEl.waitForDisplayed();
		await surfaceEl.click();
```

File: test/specs/repo/extensions/visual-editor.js [hunk 7]
```diff
@@ -4,43 +4,49 @@ const assert = require( 'assert' );
...
-		browser.keys( 'X' );
-		$( '.oo-ui-processDialog-actions-primary a' ).click();
+		const inputEl = await $( '.oo-ui-inputWidget-input' );
+		await inputEl.waitForDisplayed();
+		await browser.keys( 'X' );
 
-		$( '.mw-parser-output' ).waitForDisplayed();
-		const contentBody = $( '.mw-parser-output' ).getText();
+		const primaryAEl = await $( '.oo-ui-processDialog-actions-primary a' );
+		await primaryAEl.click();
```
Comment (rti) [inline test/specs/repo/extensions/visual-editor.js:44]:
```suggestion
		const primaryAEl = await $( '.oo-ui-processDialog-actions-primary a' );
		await primaryAEl.waitForDisplayed();
		await primaryAEl.click();
```

File: test/specs/repo/extensions/wikibase-local-media.js [hunk 8]
```diff
@@ -3,53 +3,52 @@
...
+		await fileUpload.waitForDisplayed();
 
-		const fileUpload = $( '#wpUploadFile' );
 		const filePath = path.join( __dirname, '/image.png' );
-		fileUpload.setValue( filePath );
+		await fileUpload.setValue( filePath );
 
-		$( 'input.mw-htmlform-submit' ).click();
+		const submitButtonEl = await $( 'input.mw-htmlform-submit' );
+		await submitButtonEl.click();
```
Comment (rti) [inline test/specs/repo/extensions/wikibase-local-media.js:27]:
```suggestion
		const submitButtonEl = await $( 'input.mw-htmlform-submit' );
		await submitButtonEl.waitForDisplayed();
		await submitButtonEl.click();
```

File: test/specs/repo/extensions/wikibase-local-media.js [hunk 9]
```diff
@@ -58,22 +57,22 @@ describe( 'WikibaseLocalMedia', function () {
...
+		itemId = await WikibaseApi.createItem( 'image-test', data );
 
-		browser.url( process.env.MW_SERVER + '/wiki/Item:' + itemId );
-		$( '.wikibase-snakview-value img' ).waitForDisplayed();
-		const imageSource = $( '.wikibase-snakview-value img' ).getAttribute( 'src' );
+		await browser.url( process.env.MW_SERVER + '/wiki/Item:' + itemId );
+		const snakviewEl = await $( '.wikibase-snakview-value img' );
+		await snakviewEl.waitForDisplayed();
+		const imageSourceEl = await $( '.wikibase-snakview-value img' );
+		const imageSource = await imageSourceEl.getAttribute( 'src' );
```
Comment (rti) [inline test/specs/repo/extensions/wikibase-local-media.js:74]:
```suggestion
		const imageSourceEl = await $( '.wikibase-snakview-value img' );
		await imageSourceEl.waitForDisplayed();
		const imageSource = await imageSourceEl.getAttribute( 'src' );
```

File: test/specs/repo/special-version.js [hunk 10]
```diff
@@ -4,16 +4,18 @@ const assert = require( 'assert' );
...
+			this.skip();
 		}
 
-		browser.url( process.env.MW_SERVER + '/wiki/Special:Version' );
-		const text = $( '#sv-software' ).getText();
-		assert.strictEqual( text.includes( 'MediaWiki\t' + process.env.MEDIAWIKI_VERSION ), true );
-
+		await browser.url( process.env.MW_SERVER + '/wiki/Special:Version' );
+		const softwareEl = await $( '#sv-software' );
+		const text = await softwareEl.getText();
```
Comment (rti) [inline test/specs/repo/special-version.js:14]:
```suggestion
		const softwareEl = await $( '#sv-software' );
		await softwareEl.waitForDisplayed();
		const text = await softwareEl.getText();
```

File: test/specs/repo_client/interwiki-links.js [hunk 11]
```diff
@@ -1,33 +1,38 @@
...
 
 		const config = {
 			user: process.env.DB_USER,
 			pass: process.env.DB_PASS,
 			database: 'client_wiki'
 		};
-		const clientLink = fs.readFileSync( __dirname + '/interwiki-link.sql', 'utf8' )
+		const clientLink = repoLinkFile
 			.replace( /<WIKI_ID>/g, 'my_wiki' )
 			.replace( /<HOSTNAME>/g, process.env.MW_SERVER );
```
Comment (rti) [inline test/specs/repo_client/interwiki-links.js:30]:
```suggestion
		const clientLink = repoLink;
```

**Review by rti — CHANGES_REQUESTED** [review]
Wow! Nice work. Thanks a lot for that.

Note: For easier review it might be helpful to split reformatting and refactoring into two different PRs next time.

File: test/specs/repo_client/interwiki-links.js (hunk 11 above)
Comment (RickiJay-WMDE) [reply test/specs/repo_client/interwiki-links.js:30]: I believe we want these values properly replaced in the SQL code

**Review by rti — APPROVED** [review]
🥳🥳🥳
👯👯👯

**Pull request merged**